.github/workflows/on_build_do_test.yml

```diff
@@ -119,12 +119,32 @@ jobs:
           HIBERNATE_MATRIX: ${{ github.event.workflow_run.event == 'push' && '-Phibernate-matrix' || '' }}
         id: maven-test
         shell: bash
-        run: >
-            cd test_dir/infinispan &&
-            ./mvnw verify -V -B -e -DrerunFailingTestsCount=2
-            -Dmaven.test.failure.ignore=true -Dansi.strip=true -Pnative,coverage $HIBERNATE_MATRIX
-            -Dorg.infinispan.test.server.extension.libs=org.jacoco:org.jacoco.agent:0.8.12:runtime
-            -Dgithub.action=true
+        run: |
+            cd test_dir/infinispan
+            ./mvnw verify -V -B -e -DrerunFailingTestsCount=2 \
+            -Dmaven.test.failure.ignore=true -Dansi.strip=true -Pnative,coverage $HIBERNATE_MATRIX \
+            -Dorg.infinispan.test.server.extension.libs=org.jacoco:org.jacoco.agent:0.8.12:runtime \
+            -Dgithub.action=true &
+            MVN_PID=$!
+            # Wait up to 4 hours (14400 seconds)
+            SECONDS=0
+            while kill -0 $MVN_PID 2>/dev/null; do
```

Comment:
Interesting, didn't know about `kill -0` as a no-op that just checks for the PID